### DISCUSSION TITLE
Supabase client

### DISCUSSION THREAD

**Discussion by barbinbrad** [rphlmr/supa-fly-stack#55] [Q&A]
Hey, thanks for the work.

I noticed that you had a Supabase client provider in V1 that would refresh the token and that you later removed it. I was wondering if you could give a short explanation on why?

**Comment by rphlmr** [comment]
Hello,

I removed It because It required some rewriting to follow Supabase JS SDK v2 changes.
I have migrated at work and will push a PR to add It again.

Now there is an [official Supabase auth helper](https://supabase.com/docs/guides/auth/auth-helpers/remix), I balance between using It or keep providing an alternative way (doing things 100% server side).
I have to try It on my work project before taking a decision 😬

**Reply by rphlmr** [reply]
Pull request is here :) https://github.com/rphlmr/supa-fly-stack/pull/57

**Reply by barbinbrad** [reply]
So I see you decided to keep your own method. I think that's a much cleaner approach personally. That's cool that you get to work with this stack at work, and even cooler that you take the time to give back. Thank you!

**Reply by rphlmr** [reply]
I have trouble with the official package relating to the httpOnly:false cookie :/

Maybe it's just me, asking myself too many questions about security.

This stack is not perfect, I should take time to write good documentation outside of code comments too.

I try as much as possible to bypass `channel.socket.setAuth()` (to be able to use RLS with Realtime, based on authenticated role), but found nothing reliable. Other solutions caused desynchronization of the access token (+ unstable behavior).

I use all this code in production and It's a pleasure to share It :) 

_Maybe I'll try to make a package. But I need time to think about the API and allowed options_

Thank you for your feedback ;)
